### PR TITLE
adding better logging for express responses that aren't successful

### PR DESCRIPTION
**Pull request by mboudreau**

### CODE REVIEW

File: src/frontend/app.ts [hunk 1]
```diff
@@ -97,6 +97,17 @@ export default (octokitApp: App): Express => {
 
 	app.use(logMiddleware);
 
+	// Catch non successful responses
+	app.use((req, res, next) => {
+		res.once("finish", () => {
+			// if status is under 200 or higher/equal to 400, but not 503 in maintenance mode, log it to figure out issues
+			if ((res.statusCode < 200 || res.statusCode >= 400) && !(res.statusCode === 503 && booleanFlag(BooleanFlags.MAINTENANCE_MODE, false))) {
+				req.log.warn({ res, req }, `Returning HTTP response of '${res.statusCode}' for path '${req.path}'`);
```
Comment (thombergs) [inline src/frontend/app.ts:105]: Maybe not in this PR, but later: we should probably expose this as a metric with the path and status as tags, so we can create a chart on top of this.

**Review by thombergs — APPROVED** [review]

File: src/frontend/app.ts [hunk 2]
```diff
@@ -97,6 +97,17 @@ export default (octokitApp: App): Express => {
 
 	app.use(logMiddleware);
 
+	// Catch non successful responses
+	app.use((req, res, next) => {
+		res.once("finish", () => {
+			// if status is under 200 or higher/equal to 400, but not 503 in maintenance mode, log it to figure out issues
+			if ((res.statusCode < 200 || res.statusCode >= 400) && !(res.statusCode === 503 && booleanFlag(BooleanFlags.MAINTENANCE_MODE, false))) {
```
Comment (KAbakumovAtl) [inline src/frontend/app.ts:104]: Do we still need maintenance mode?

File: src/jira/client/index.ts [hunk 3]
```diff
@@ -274,9 +274,9 @@ export default async (
  */
 const batchedBulkUpdate = async (
 	data,
-	options,
 	instance: AxiosInstance,
-	installationId: number
+	installationId: number,
+	options?: { preventTransitions: boolean }
```
Comment (KAbakumovAtl) [inline src/jira/client/index.ts:279]: I wonder is it related to logging or you are also trying to fix passing `preventTransitions` to Jira in this PR?

File: src/frontend/app.ts (hunk 2 above)
Comment (mboudreau) [reply src/frontend/app.ts:104]: It's to ignore maintenance mode 503s

File: src/jira/client/index.ts (hunk 3 above)
Comment (mboudreau) [reply src/jira/client/index.ts:279]: Not related to logging, was just fixing a type

File: src/frontend/app.ts (hunk 2 above)
Comment (KAbakumovAtl) [reply src/frontend/app.ts:104]: Yes, but would we need it at all? We've already performed the migration, so eventually we'll cleanup the flag.

**Review by KAbakumovAtl — APPROVED** [review]

**Pull request merged**